static/app/components/alerts/snoozeAlert.tsx

```diff
@@ -141,22 +154,31 @@ function SnoozeAlert({isSnoozed, onSnooze, projectSlug, ruleId, hasAccess}: Prop
         size="sm"
         icon={<IconSound />}
         disabled={disabled || !hasAccess}
-        onClick={() => handleMute('me')}
+        hasDropdown={ruleActionCategory !== RuleActionsCategories.NoDefault}
+        onClick={() => {
+          const target =
+            ruleActionCategory !== RuleActionsCategories.AllDefault ? 'everyone' : 'me';
+          handleMute(target);
+        }}
       >
-        {t('Mute')}
+        {ruleActionCategory === RuleActionsCategories.AllDefault
+          ? t('Mute for me')
+          : t('Mute for everyone')}
       </MuteButton>
-      <DropdownMenu
-        size="sm"
-        trigger={triggerProps => (
-          <DropdownTrigger
-            {...triggerProps}
-            aria-label={t('Mute alert options')}
-            icon={<IconChevron direction="down" size="xs" />}
-          />
-        )}
-        items={dropdownItems}
-        isDisabled={disabled || !hasAccess}
-      />
+      {ruleActionCategory !== RuleActionsCategories.NoDefault && (
```

Comment:
nit: could simplify the logic a bit if you wanted and do 

`const hasDropdown = dropdownItems.filter(item => !item.hidden).length > 0`